monitor_ingest/actions/setup_entity.py

```diff
@@ -0,0 +1,98 @@
+import yaml
+from st2common.runners.base_action import Action
+from mam.sdk import (entitytype, constants, kpifunction, dimension)
+
+__all__ = ['SetupEntityAction']
+
+
+class SetupEntityAction(Action):
+    def __init__(self, config):
+        super(SetupEntityAction, self).__init__(config)
+        self._config = self.config
+        self._credentials = self._config.get('credentials', None)
+        self._data_file_path = self._config.get('data_file_path', None)
+        self._action_type = self._config.get('action_type', None)
+        
+        if not self._config:
+            raise ValueError('Missing config yaml')
+        if not self._credentials:
+            raise ValueError('Missing IBM Watson IoT credentials in config file')
+        if not self._data_file_path:
+            raise ValueError('Missing data file path in config file')
+        if not self._action_type:
+            raise ValueError('Missing action type key in config file')
+
+    def run(self):
+        operations_completed = {}
+        entitytype_data_path = None
+        constants_data_path = None
+        dimension_data_path = None
+        function_data_path = None
+        
+        if self._action_type == "SetupEntityAction":
```

Comment:
curious why these are not parameters to the action and instead hard coded in the config?

Reply:
our config file is actually working as a handler to perform different actions on the entity. We had updated the code to follow a modular approach now. Instead of if loops  we tried using functions to call a particular action type.